src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Dsa.cs

```diff
@@ -73,8 +73,24 @@ internal static int DsaKeySize(SafeDsaHandle dsa)
         [return: MarshalAs(UnmanagedType.Bool)]
         private static extern bool DsaSign(SafeDsaHandle dsa, ref byte hash, int hashLength, ref byte refSignature, out int outSignatureLength);
 
-        internal static bool DsaVerify(SafeDsaHandle dsa, ReadOnlySpan<byte> hash, int hashLength, ReadOnlySpan<byte> signature, int signatureLength) =>
-            DsaVerify(dsa, ref MemoryMarshal.GetReference(hash), hashLength, ref MemoryMarshal.GetReference(signature), signatureLength);
+        internal static bool DsaVerify(SafeDsaHandle dsa, ReadOnlySpan<byte> hash, ReadOnlySpan<byte> signature)
+        {
+            bool ret = DsaVerify(
+                dsa,
+                ref MemoryMarshal.GetReference(hash),
+                hash.Length,
+                ref MemoryMarshal.GetReference(signature),
+                signature.Length);
+
+            if (!ret)
+            {
+                // OpenSSL DSA signature processing requires a DER encode and decode, so
+                // the error queue may have been contaminated.
+                ErrClearError();
```

Comment:
It looks like we're clearing the error queue in the native code:
https://github.com/dotnet/corefx/blob/f95b511d7c24d0d8033fd08d695c185a1b560ec9/src/Native/Unix/System.Security.Cryptography.Native/pal_dsa.cpp#L90-L109
Do we need both?

Reply:
When I noticed the potential for contamination I had jumped straight to DSA_verify, hadn't noticed we fixed it up here.  So, yeah, I agree that we don't need this one.

Reply:
I'm removing the double clean up but keeping the new signature for the method.